### ISSUE TITLE
llvm-config usage in cross compilation?

### ISSUE THREAD

**Issue by kanavin** [llvm/llvm-project#58984]
When cross compiling items such as mesa with llvm support, the build process relies on output of llvm-config, and therefore on ability to run llvm-config. But if llvm as a whole has been cross-built, then it is not possible to run llvm-config because it needs a different computer.

How should this situation be handled? One option is to make llvm-config a arch-independent script (it is now a cpp program). Another is to also make it a script, but have it be only a thin wrapper over llvm's pkg-config file (which currently does not exist).

Suggestions?

Yocto project has until now dealt with this by 
- building llvm-config first for the build machine, then running that. This does not work so well due to mismatches between host and target configurations.
- providing own replacement for llvm-config as a shell script. This also isn't great as its output is not kept in sync with what actual llvm-config produces.

**Comment by tstellar** [comment]
Internally, within the LLVM sub-projects, we have been migrating away from llvm-config and instead relying on the installed cmake files to provide the necessary information.  Would you be able to do the same?

**Comment by eli-schwartz** [comment]
> Internally, within the LLVM sub-projects, we have been migrating away from llvm-config and instead relying on the installed cmake files to provide the necessary information. Would you be able to do the same?

Does this mean that LLVM has officially dropped support for being used in projects that choose a build system other than cmake?

If not, then how should projects that use a build system other than cmake get information out of a cmake program?

**Comment by tstellar** [comment]
> > Internally, within the LLVM sub-projects, we have been migrating away from llvm-config and instead relying on the installed cmake files to provide the necessary information. Would you be able to do the same?
> 
> Does this mean that LLVM has officially dropped support for being used in projects that choose a build system other than cmake?
> 
No, but for something like cross-compiling, using the CMake files is going to be much easier than trying to build a working cross-compiled llvm-config.

**Comment by dcbaker** [comment]
Would you take patches to add pkg-config files? That would probably solve the issue entirely for everyone who's not using cmake, maybe even removing the need for llvm-config altogether.

**Comment by tstellar** [comment]
> Would you take patches to add pkg-config files? That would probably solve the issue entirely for everyone who's not using cmake, maybe even removing the need for llvm-config altogether.

I'm not sure how others feel, but I am in favor of this.

**Comment by pogo59** [comment]
Seems like tblgen would be in the same boat--it needs to run on the build host, not the target. Do we get that right? If so, could llvm-config be handled the same way?  If not, that's its own issue.
